staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -592,6 +598,19 @@ func (o *DebugOptions) generatePodCopyWithDebugContainer(pod *corev1.Pod) (*core
 			c.Args = nil
 		}
 	}
+	if len(o.Capabilities) > 0 {
+		fmt.Printf("o.Capabilities is %v \n ", o.Capabilities)
+		capStrs := strings.Split(o.Capabilities, ",")
+		if len(capStrs) > 0 {
+			fmt.Printf("capStrs is %v \n ", capStrs)
+			// TODO: Need to validate whether Capabilities comply with established rules
+			c.SecurityContext = &corev1.SecurityContext{
```

Comment:
this stomps any existing security context (and I didn't think security contexts could be set in ephemeral containers currently)

Reply:
That's correct, though this method isn't used with ephemeral containers and is modifying a fresh copy of the pod.